iface/get.go

```diff
@@ -2,79 +2,159 @@ package iface
+// "go:mock" directive, returning text-template-friendly representations.
+func GetAllInterfaces(pkgs []*packages.Package) ([]Interface, error) {
+	var allInfo []ifaceInfo
+	for _, pkg := range pkgs {
+		for _, fileNode := range pkg.Syntax {
+			ast.Inspect(fileNode, func(node ast.Node) bool {
+				// Only consider declarations with godoc comments.
+				decl, isGenDecl := node.(*ast.GenDecl)
+				if !isGenDecl || decl.Doc == nil {
+					return true
+				}
+				// Look for the first go:mock directive in the comments.
+				for _, comment := range decl.Doc.List {
+					args := strings.TrimPrefix(comment.Text, "//go:mock")
+					if args == comment.Text {
+						continue
+					}
+
+					// Build a qualified output pathname based on the output
+					// filename (or a default) and the input filepath.
+					var (
+						inputPath  = pkg.Fset.File(fileNode.Pos()).Name()
+						outputFile = strings.TrimSpace(args)
+						outputPath string
+					)
+					if outputFile == "" {
+						outputPath = strings.TrimSuffix(inputPath, ".go") + "_mock.go"
+					} else {
+						outputPath = filepath.Join(filepath.Dir(inputPath), outputFile)
+					}
 
-	if len(pkgs) < 1 {
-		return Interface{}, errors.Wrap(err, "failed to find/load package info")
-	} else if len(pkgs) > 1 {
-		return Interface{}, errors.Wrap(err, "found more than one matching package")
+					for _, spec := range decl.Specs {
+						// Only consider type declarations.
+						spec, isType := spec.(*ast.TypeSpec)
+						if !isType {
+							continue
+						}
+						// Get the type info for this declaration.
+						if object := pkg.TypesInfo.Defs[spec.Name]; object != nil {
+							allInfo = append(allInfo, ifaceInfo{
+								pkg:        pkg,
+								fileNode:   fileNode,
+								object:     object,
+								outputPath: outputPath,
+							})
+							return true
+						}
+					}
+				}
+				return true
+			})
+		}
 	}
-	pkg := pkgs[0]
-
-	// Keep track of each file's imports, along with their name (if renamed)
-	fileImps := map[token.Pos][]Import{}
-	for _, fileAST := range pkg.Syntax {
-		var imps []Import
-		for _, fileImp := range fileAST.Imports {
-			imp := Import{
-				Path: strings.Trim(fileImp.Path.Value, "\""),
-			}
-			if fileImp.Name != nil {
-				imp.Name = fileImp.Name.Name
-			}
-			imps = append(imps, imp)
+
+	var ifaces []Interface
+	for _, info := range allInfo {
+		iface, ifaceErr := newInterface(info)
+		if ifaceErr != nil {
+			return nil, ifaceErr
 		}
-		fileImps[pkg.Fset.File(fileAST.Pos()).Pos(0)] = imps
+		ifaces = append(ifaces, iface)
 	}
+	return ifaces, nil
+}
 
+// GetInterface searches the given package for the given interface and returns
+// its text-template-friendly representation.
+func GetInterface(pkg *packages.Package, ifaceName string, outputFile string) (Interface, error) {
 	// Find the interface by name
 	ifaceObj := pkg.Types.Scope().Lookup(ifaceName)
 	if ifaceObj == nil {
-		return Interface{}, errors.Errorf("interface not found in package: %s", ifaceName)
```

Comment:
I didn't know there used to be a [third-party (?) `errors` package](https://github.com/pkg/errors) before `errors` and `fmt.Errorf` were invented!